# Covid Dataset for India 
Kaggle sourced public dataset


libraries

In [2]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
import os

In [3]:
ds = pd.read_csv('covid_19_india.csv',index_col=0)
ds["ConfirmedIndianNational"].replace({"-": "0"}, inplace=True)
ds["ConfirmedForeignNational"].replace({"-": "0"}, inplace=True)
ds['ConfirmedIndianNational'] = pd.to_numeric(ds['ConfirmedIndianNational'])
ds['ConfirmedForeignNational'] = pd.to_numeric(ds['ConfirmedForeignNational'])
print(ds.head())
print(ds.shape)

         Date     Time State/UnionTerritory  ConfirmedIndianNational  \
Sno                                                                    
1    30/01/20  6:00 PM               Kerala                        1   
2    31/01/20  6:00 PM               Kerala                        1   
3    01/02/20  6:00 PM               Kerala                        2   
4    02/02/20  6:00 PM               Kerala                        3   
5    03/02/20  6:00 PM               Kerala                        3   

     ConfirmedForeignNational  Cured  Deaths  Confirmed  
Sno                                                      
1                           0      0       0          1  
2                           0      0       0          1  
3                           0      0       0          2  
4                           0      0       0          3  
5                           0      0       0          3  
(894, 8)


894 data points sorted over 74 days (length of dates) exciting right ?

In [4]:
states = ds['State/UnionTerritory'].unique()
print(states)
dates = ds['Date'].unique()
print(len(dates))


['Kerala' 'Telengana' 'Delhi' 'Rajasthan' 'Uttar Pradesh' 'Haryana'
 'Ladakh' 'Tamil Nadu' 'Karnataka' 'Maharashtra' 'Punjab'
 'Jammu and Kashmir' 'Andhra Pradesh' 'Uttarakhand' 'Odisha' 'Puducherry'
 'West Bengal' 'Chhattisgarh' 'Chandigarh' 'Gujarat' 'Himachal Pradesh'
 'Madhya Pradesh' 'Bihar' 'Manipur' 'Mizoram'
 'Andaman and Nicobar Islands' 'Goa' 'Unassigned' 'Assam' 'Jharkhand'
 'Arunachal Pradesh' 'Tripura']
74


In [5]:
res = []
for i in range(0,len(states),1) :  
 print( ds.loc[ds['State/UnionTerritory'] == states[i]])
   


         Date     Time State/UnionTerritory  ConfirmedIndianNational  \
Sno                                                                    
1    30/01/20  6:00 PM               Kerala                        1   
2    31/01/20  6:00 PM               Kerala                        1   
3    01/02/20  6:00 PM               Kerala                        2   
4    02/02/20  6:00 PM               Kerala                        3   
5    03/02/20  6:00 PM               Kerala                        3   
..        ...      ...                  ...                      ...   
755  08/04/20  5:00 PM               Kerala                        0   
786  09/04/20  5:00 PM               Kerala                        0   
817  10/04/20  5:00 PM               Kerala                        0   
848  11/04/20  5:00 PM               Kerala                        0   
879  12/04/20  5:00 PM               Kerala                        0   

     ConfirmedForeignNational  Cured  Deaths  Confirmed  
Sno  

618                         0      0       0         77  
         Date     Time State/UnionTerritory  ConfirmedIndianNational  \
Sno                                                                    
532  01/04/20  7:30 PM                Assam                        0   
561  02/04/20  6:00 PM                Assam                        0   
591  03/04/20  6:00 PM                Assam                        0   
622  04/04/20  6:00 PM                Assam                        0   
652  05/04/20  6:00 PM                Assam                        0   
682  06/04/20  6:00 PM                Assam                        0   
712  07/04/20  6:00 PM                Assam                        0   
743  08/04/20  5:00 PM                Assam                        0   
774  09/04/20  5:00 PM                Assam                        0   
805  10/04/20  5:00 PM                Assam                        0   
836  11/04/20  5:00 PM                Assam                        0   
867  1

In [9]:
#lets analyse state by state
#first state kerala
kerala_data = ds.loc[ds['State/UnionTerritory'] == 'Kerala',]

k_data_cont = []
#making data continous total number of dates till date for every date
#lets have some fun
for i in range(len(kerala_data)):
    k_data_cont.append(kerala_data.iloc[:i,3].sum())
        
print(len(k_data_cont ))
print(k_data_cont)


74
[0, 1, 2, 4, 7, 10, 13, 16, 19, 22, 25, 28, 31, 34, 37, 40, 43, 46, 49, 52, 55, 58, 61, 64, 67, 70, 73, 76, 79, 82, 85, 88, 91, 94, 97, 100, 103, 106, 109, 117, 126, 141, 158, 175, 194, 213, 235, 258, 282, 307, 332, 358, 391, 436, 496, 583, 684, 794, 923, 1091, 1091, 1091, 1091, 1091, 1091, 1091, 1091, 1091, 1091, 1091, 1091, 1091, 1091, 1091]


In [13]:
%matplotlib inline
plt.xlabel('active cases')
plt.ylabel('days')
plt.legend('kerala')
plt.plot(dates,k_data_cont)

SyntaxError: invalid syntax (<ipython-input-13-61f67c92c13c>, line 5)

## Country Wide?
country wide data needs to be sorted data wise with some dates having more than 1 point
writing a for loop


In [8]:
d_count = []
for i in range(len(dates)):
    d_count.append(ds.iloc[:i,3].sum())
        


